plugins/runner/runner.go

```diff
@@ -34,35 +34,36 @@ import (
 const defaultPollInterval = 2 * time.Second
 
 type pluginRunner struct {
-	config           types.PluginConfig
-	containerManager containermanager.PluginContainerManager
-	client           runnerclient.ClientWithResponsesInterface
+	config         types.PluginConfig
+	runtimeHandler runtimehandler.PluginRuntimeHandler
+	client         runnerclient.ClientWithResponsesInterface
 }
 
 func New(ctx context.Context, config types.PluginConfig) (types.PluginRunner, error) {
 	// Create docker container
 	// TODO: switch to factory once the support for more container engines is added
-	manager, err := docker.New(ctx, config)
+	//handler, err := docker.New(ctx, config)
+	handler, err := binary.New(ctx, config)
```

Comment:
Note: we need to create handler according to the config (whether binary mode is enabled or not)